bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/ThingManager.java

```diff
@@ -374,6 +306,91 @@ public void handlerRemoved(Thing thing, ThingHandler thingHandler) {
     }
 
     @Override
+    public void changeThingType(final Thing thing, final ThingTypeUID thingTypeUID,
+            final Configuration configuration) {
+        scheduler.schedule(new Runnable() {
+            @Override
+            public void run() {
+                ThingUID thingUID = thing.getUID();
+                waitForRunningHandlerRegistrations(thingUID);
+                ThingType thingType = thingTypeRegistry.getThingType(thingTypeUID);
+                if (thingType == null) {
```

Comment:
i would prefer if we check this outside the runnable implementation and throw an exception if the thing type does not exist... then we should document this exception for the service and for the callback